transform/mattermost-analytics/models/intermediate/product/active_users/int_user_active_days_spined.sql

```diff
@@ -17,9 +17,13 @@ with user_active_days as (
         coalesce(s.server_id, m.server_id, l.server_id) as server_id,
         coalesce(s.user_id, m.user_id, l.user_id) as user_id,
         coalesce(s.is_active, m.is_active, l.is_active) as is_active,
-        s.server_id is not null as is_desktop_or_server,
+        case when s.server_id is not null and s.client_type = 'Desktop' then true else false end as is_client_desktop,        
+        case when s.server_id is not null and s.client_type = 'Webapp' then true else false end as is_client_webapp,
```

Comment:
Nulls are not handled here. For example if `server_id` is null and `client_type` <> null, result will be false. Another example is if `client_type` is null (result will be null).

Reply:
the `server_id` can only be null due to the full outer join, which indicates that the `server` doesn't belong in the specific source (client/legacy/mobile). We have this filter in the upstream int model. If the `client_type` is null, I am handling that in the `unknown` bucket as discussed.

https://github.com/mattermost/mattermost-data-warehouse/blob/e98062c60d8a087eb067815e5ed2b302ded49c0c/transform/mattermost-analytics/models/intermediate/product/active_users/int_user_active_days_server_telemetry.sql#L24